src/lib/homestore_backend/hs_homeobject.hpp

```diff
@@ -45,51 +45,80 @@ class HSHomeObject : public HomeObjectImpl {
         peer_id_t replica_set_uuid;
         pg_members members[1]; // ISO C++ forbids zero-size array
     };
+
+    struct shard_info_superblk {
+        shard_id_t id;
+        pg_id_t placement_group;
+        ShardInfo::State state;
+        uint64_t created_time;
+        uint64_t last_modified_time;
+        uint64_t available_capacity_bytes;
+        uint64_t total_capacity_bytes;
+        uint64_t deleted_capacity_bytes;
+        homestore::chunk_num_t chunk_id;
+    };
 #pragma pack()
 
     struct HS_PG : public PG {
         homestore::superblk< pg_info_superblk > pg_sb_;
         shared< homestore::ReplDev > repl_dev_;
-
+        std::optional< homestore::chunk_num_t > any_allocated_chunk_id_{};
```

Comment:
where will this be used, want to get the pdev_id here?
it is a little wired

Reply:
no, just to cache any chunk_id from random shard for pg.  thus it can use it to call chunk selector->chunk_to_hints() and no need to list shard from the pg any more for per shard create request

Reply:
chunk_to_hint() is used to get pdev_id , why not just save pdev_id here

Reply:
I want to do it before as it is more straightforward. but ShardManager only know the shard to chunk map and not-aware of pdev id at all,  so it cache chunk id here and leave the chunk to pdev map to heap chunk selector.